compiler/cpp/src/thrift/generate/t_cpp_generator.cc

```diff
@@ -693,55 +705,58 @@ void t_cpp_generator::generate_enum_to_string_helper_function(std::ostream& out,
 void t_cpp_generator::generate_consts(std::vector<t_const*> consts) {
   string f_consts_name = get_out_dir() + program_name_ + "_constants.h";
   ofstream_with_content_based_conditional_update f_consts;
-  f_consts.open(f_consts_name);
+  if (consts.size() > 0)
+  {
+    f_consts.open(f_consts_name);
 
-  string f_consts_impl_name = get_out_dir() + program_name_ + "_constants.cpp";
-  ofstream_with_content_based_conditional_update f_consts_impl;
-  f_consts_impl.open(f_consts_impl_name);
+    string f_consts_impl_name = get_out_dir() + program_name_ + "_constants.cpp";
+    ofstream_with_content_based_conditional_update f_consts_impl;
+    f_consts_impl.open(f_consts_impl_name);
 
-  // Print header
-  f_consts << autogen_comment();
-  f_consts_impl << autogen_comment();
+    // Print header
+    f_consts << autogen_comment();
+    f_consts_impl << autogen_comment();
 
-  // Start ifndef
-  f_consts << "#ifndef " << program_name_ << "_CONSTANTS_H" << endl << "#define " << program_name_
-           << "_CONSTANTS_H" << endl << endl << "#include \"" << get_include_prefix(*get_program())
-           << program_name_ << "_types.h\"" << endl << endl << ns_open_ << endl << endl;
+    // Start ifndef
+    f_consts << "#ifndef " << program_name_ << "_CONSTANTS_H" << endl << "#define " << program_name_
+             << "_CONSTANTS_H" << endl << endl << "#include \"" << get_include_prefix(*get_program())
+             << program_name_ << "_types.h\"" << endl << endl << ns_open_ << endl << endl;
```

Comment:
What's the point about all those whitespace changes?

Also, CI fails largely: https://api.travis-ci.org/v3/job/672800329/log.txt

Reply:
Hi @Jens-G , because all those statements is under the condition ` if (consts.size() > 0)`, and I want to follow the current coding style  by Indent two spaces. BTW, I don't know why the CI is always fails , even the change is very small just like add documents etc. I built it through Makefile and CMake are both OK in my local.